code/modules/cm_tech/implements/armor.dm

```diff
@@ -184,3 +187,196 @@
+				text += "to DYNAMIC. Overdose subsystems will inject chemicals but will not go above overdose levels."
+			else
+				text += "to STRICT. Overdose subsystems will refuse to inject if any of chemicals will overdose."
+		to_chat(user, SPAN_NOTICE(text))
+		return TRUE
+	return
+
+/obj/item/clothing/accessory/health/research_plate/emergency_injector/can_recycle(mob/living/user)
+	if(used)
+		return TRUE
+	return FALSE
+
+/obj/item/clothing/accessory/health/research_plate/emergency_injector/on_attached(obj/item/clothing/S, mob/living/carbon/human/user)
+	. = ..()
+	wearer = user
+	activation = new /datum/action/item_action/emergency_plate/inject_chemicals(src, attached_uni)
+	activation.give_to(wearer)
+
+/obj/item/clothing/accessory/health/research_plate/emergency_injector/on_removed(mob/living/user, obj/item/clothing/C)
+	. = ..()
+	qdel(activation)
+	attached_uni = null
+
+/obj/item/clothing/accessory/health/research_plate/emergency_injector/on_removed_sig(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(. == FALSE)
+		return
+	qdel(activation)
+	attached_uni = null
+
+//Action buttons
+/datum/action/item_action/emergency_plate/inject_chemicals/New(Target, obj/item/holder)
+	. = ..()
+	name = "Inject Emergency Plate"
+	action_icon_state = "regular2_100"
+	button.name = name
+	button.overlays.Cut()
+	button.overlays += image('icons/obj/items/items.dmi', button, action_icon_state)
+
+/obj/item/clothing/accessory/health/research_plate/emergency_injector/ui_action_click(mob/owner, obj/item/holder)
+	if(used)
+		to_chat(wearer, SPAN_DANGER("The plate inner reserve is empty, replace the plate!"))
+		return
+	for(var/chemical in chemicals_to_inject)
+		var/datum/reagent/reag = GLOB.chemical_reagents_list[chemical]
+		if(wearer.reagents.get_reagent_amount(chemical) + chemicals_to_inject[chemical] > reag.overdose)
+			if(od_protection_mode == EMERGENCY_PLATE_OD_PROTECTION_STRICT)
+				to_chat(wearer, SPAN_DANGER("You hold the two buttons, but the plate buzzes and refuses to inject, indicating the potential overdose!"))
+				return
+			else if (od_protection_mode == EMERGENCY_PLATE_OD_PROTECTION_DYNAMIC)
+				var/adjust_volume_to_inject = reag.overdose - wearer.reagents.get_reagent_amount(chemical)
+				chemicals_to_inject[chemical] = adjust_volume_to_inject
+				warning_type = 2
+		if(wearer.reagents.get_reagent_amount(chemical) + chemicals_to_inject[chemical] > reag.overdose && od_protection_mode == EMERGENCY_PLATE_OD_PROTECTION_OFF)
+			warning_type = 1
+		wearer.reagents.add_reagent(chemical, chemicals_to_inject[chemical])
+	if(warning_type == 1)
+		to_chat(wearer, SPAN_DANGER("You hold the two buttons, and the plate injects the chemicals, but makes a worrying beep, indicating overdose!"))
+	if(warning_type == 2)
+		to_chat(wearer, SPAN_DANGER("You hold the two buttons, and the plate injects the chemicals, but makes a reliefing beep, indicating it adjusted amounts it injected to prevent overdose!"))
```

Comment:
```suggestion
		to_chat(wearer, SPAN_DANGER("You hold the two buttons, and the plate injects the chemicals, but makes a relieving beep, indicating it adjusted amounts it injected to prevent overdose!"))
```
i presume this was what was intended, but i'm not entirely sure about what a relieving beep is, mind you